src/main/java/ru/skypro/homework/controller/CommentsController.java

```diff
@@ -0,0 +1,86 @@
+package ru.skypro.homework.controller;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import ru.skypro.homework.config.MapperUtil;
+import ru.skypro.homework.dto.comments.CommentsDTO;
+import ru.skypro.homework.dto.comments.CreateOrUpdateCommentDTO;
+import ru.skypro.homework.entity.Ad;
+import ru.skypro.homework.entity.Comments;
+import ru.skypro.homework.service.impl.AdsServiceImpl;
+import ru.skypro.homework.service.interfaces.CommentsService;
+
+import java.util.List;
+
+@Slf4j //  добавляет логгер в класс
+@CrossOrigin(value = "http://localhost:3000") // позволяет настроить CORS (Cross-Origin Resource Sharing)
+// для данного контроллера. Указывает, что этот контроллер может обрабатывать запросы с указанного домена
+// (http://localhost:3000), даже если он отличается от домена, на котором запущено приложение.
+@RequiredArgsConstructor // генерирует конструктор с аргументами для всех полей, помеченных аннотацией @NonNull
+@RestController
+@RequestMapping("/ads/{adId}/comments")
+@Tag(name = "Комментарии")
+public class CommentsController {
+
+    private CommentsService commentsService;
+
+//    private AdsService adsService;
+    private AdsServiceImpl adsService;
+
+    private MapperUtil mapperUtil;
+
+    public CommentsController(CommentsService commentsService, AdsServiceImpl adsService, MapperUtil mapperUtil) {
```

Comment:
```suggestion
    public CommentsController(CommentsService commentsService, AdsService adsService, MapperUtil mapperUtil) {
```
инъекцию надо делать по интерфейсу, а не реализации